### PR TITLE
Adds a cooldown to PDA send to all messages.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]:

:cl: Dax Dupont
balance: Adds a cooldown to PDA send to all messages.
fix: Last sent message cooldown actually works now.
/:cl:

[why]: People using it to spam is getting annoying, a cooldown seems appropriate considering it sends messages to everyone.

### CODE REVIEW

**Comment by QualityVan** [comment]
despondent honking noises

**Comment by F-OS** [comment]
How long is the cooldown i can't be arsed to math

**Comment by DaxDupont** [comment]
@F-OS 3 minutes

**Comment by Therapist1** [comment]
gee, i can only imagine why this one got opened

**Comment by Bombpants** [comment]
What was it before?

**Comment by DaxDupont** [comment]
@Bombpants none

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -686,7 +690,10 @@ GLOBAL_LIST_EMPTY(PDAs)
 	add_overlay(icon_alert)
 
 /obj/item/pda/proc/send_to_all(mob/living/U)
-	send_message(U,get_viewable_pdas())
+	if (last_everyone && world.time < last_everyone + 1800)
```
Comment (ShizCalev) [inline code/game/objects/items/devices/PDA/PDA.dm]: `last_everyone + (3 MINUTES)` please

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/game/objects/items/devices/PDA/PDA.dm]: also maybe like a minute and a half pls

Comment (DaxDupont) [reply code/game/objects/items/devices/PDA/PDA.dm]: @PKPenguin321 I'll do two minutes

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -609,11 +610,11 @@ GLOBAL_LIST_EMPTY(PDAs)
 		t = Gibberish(t, 100)
 	return t
 
-/obj/item/pda/proc/send_message(mob/living/user, list/obj/item/pda/targets)
+/obj/item/pda/proc/send_message(mob/living/user, list/obj/item/pda/targets, everyone)
 	var/message = msg_input(user)
 	if(!message || !targets.len)
 		return
-	if(last_text && world.time < last_text + 5)
+	if((last_text && world.time < last_text + 10) || (everyone && last_everyone && world.time < last_everyone + (2 MINUTES)))
```
Comment (ShizCalev) [inline code/game/objects/items/devices/PDA/PDA.dm]: actually, since the time is now present in two places, make it a define please.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Anonmare** [comment]
Make sure that you can't bypass the cooldown by taking the cart out and slotting it back in or using other PDAs. Or do if that's your kink

**Comment by kevinz000** [comment]
1 minute is more sane but i'm probably just too nice

**Comment by arsserpentarium** [comment]
Make cooldown 5 mins for lawyers pda.
Add send all option to clown pda. Make cooldown 0.5 s. Honk.

**Comment by iskyp** [comment]
mute your pda it's not hard

**Comment by imsxz** [comment]
u mad whiteboi? :)

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**